pkg/target/kafka.go

```diff
@@ -243,6 +244,8 @@ func (kt *KafkaTarget) Write(messages []*models.Message) (*models.TargetWriteRes
 				if originalMessage.AckFunc != nil {
 					originalMessage.AckFunc()
 				}
+				originalMessage.TimeRequestStarted = requestStarted
+				originalMessage.TimeRequestFinished = time.Now().UTC()
```

Comment:
Is this intentional we set it only for successful response?

Reply:
So if for failures too, then maybe set it before whole if/else block? Right after `result := <-kt.asyncResults`

Reply:
Yeah good point, cleaning up